src/overlays/actors/ovl_En_Toto/z_en_toto.c

```diff
@@ -21,86 +60,703 @@ const ActorInit En_Toto_InitVars = {
+    { 0xFF64, 0x0016, 0xFE7E },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/func_80BA4B24.asm")
+static u16 D_80BA5120[] = { 67, 68, 69, 70 };
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/func_80BA4C0C.asm")
+static u8 D_80BA5128[] = { 8, 4, 2, 1 };
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/func_80BA4C44.asm")
+static EnTotoUnkFunc D_80BA512C[] = {
+    func_80BA3F2C, func_80BA3ED4, func_80BA3F2C, func_80BA3F2C, func_80BA3F2C, func_80BA3ED4,
+    func_80BA42BC, func_80BA3ED4, func_80BA44A0, func_80BA3FB0, func_80BA402C, func_80BA46D8,
+    func_80BA47E0, func_80BA49A4, func_80BA3FB0, func_80BA402C, func_80BA3F2C, func_80BA3EC0,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/func_80BA4CB4.asm")
+static EnTotoUnkFunc D_80BA5174[] = {
+    func_80BA40D4, func_80BA4128, func_80BA415C, func_80BA3EE8, func_80BA3EE8, func_80BA407C,
+    func_80BA43F4, func_80BA445C, func_80BA4530, func_80BA4204, func_80BA407C, func_80BA4740,
+    func_80BA407C, func_80BA4A00, func_80BA3FCC, func_80BA407C, func_80BA4B24,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/EnToto_Update.asm")
+static EnTotoActionFunc D_80BA51B8[] = {
+    func_80BA39C8,
+    func_80BA3CC4,
+    func_80BA3DBC,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/EnToto_Draw.asm")
+void func_80BA36C0(EnToto* this, GlobalContext* globalCtx, s32 index) {
+    this->unk2B7 = 0;
+    this->actionFuncIndex = index;
+    D_80BA501C[this->actionFuncIndex](this, globalCtx);
+}
+
+void EnToto_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnToto* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    if (globalCtx->sceneNum == 0x15 && (gSaveContext.time >= 0x4000 && gSaveContext.time < 0xE555)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 30.0f);
+    this->actor.bgCheckFlags |= 0x400;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600A978, globalCtx->sceneNum == 0x12 ? &D_06003AA8 : &D_0600C880,
+                     this->limbDrawTbl, this->transitionDrawTbl, 18);
+    func_80BA36C0(this, globalCtx, 0);
+    this->actor.shape.rot.x = 0;
+}
+
+void EnToto_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnToto* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BA383C(EnToto* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) && this->actionFuncIndex == 1 &&
+        this->skelAnime.animCurrentSeg != &D_06000C80) {
+        if (globalCtx->msgCtx.unk11F04 != 0x2A98 && globalCtx->msgCtx.unk11F04 != 0x2A99) {
+            if (this->unk2B4 & 1 || Rand_ZeroOne() > 0.5f) {
+                this->unk2B4 = (this->unk2B4 + 1) & 3;
```

Comment:
```suggestion
                this->unk2B4 = (this->unk2B4 + 1) % 4;
```
Would this work?

Reply:
Sadly it does not
![image](https://user-images.githubusercontent.com/31974442/120551835-56eb0980-c3ab-11eb-96ef-abad211d51cb.png)